samples/host_exerciser/host_exerciser.h

```diff
@@ -0,0 +1,449 @@
+    uint8_t ForcedTestCmpl : 1;
+    uint32_t Reserved : 29;
+  };
+};
+
+
+// CSR CFG
+union he_cfg {
+  enum {
+    offset = HE_CFG
+  };
+  uint64_t value;
+  struct {
+    uint8_t DelayEn : 1;
+    uint8_t Continuous : 1;
+    uint8_t TestMode : 3;
+    uint8_t ReqLen : 2;
+    uint8_t Rsvd_7 : 1;
+    uint8_t Rsvd_8 : 1;
+    uint16_t Rsvd_19_9 : 11;
+    uint8_t TestCfg : 8;
+    uint8_t IntrOnErr : 1;
+    uint8_t IntrTestMode : 1;
+    uint64_t Rsvd_63_30 : 34;
+  };
+};
+
+// CSR INACT THRESH
+union he_inact_thresh {
+  enum {
+    offset = HE_INACT_THRESH
+  };
+  uint32_t value;
+  struct {
+    uint32_t InactivtyThreshold : 32;
+  };
+};
+
+// INTERRUPT0
+union he_interrupt0 {
+  enum {
+    offset = HE_INTERRUPT0
+  };
+  uint32_t value;
+  struct {
+    uint16_t apci_id : 16;
+    uint16_t VectorNUm : 8;
+    uint16_t Rsvd_31_24 : 8;
+  };
+};
+
+// SWTEST MSG
+union he_swtest_msg {
+  enum {
+    offset = HE_SWTEST_MSG
+  };
+  uint64_t value;
+  struct {
+    uint64_t swtest_msg : 64;
+  };
+};
+
+// STATUS0
+union he_status0 {
+  enum {
+    offset = HE_STATUS0
+  };
+  uint64_t value;
+  struct {
+    uint32_t numWrites : 32;
+    uint32_t numReads : 32;
+  };
+};
+
+// STATUS1
+union he_status1 {
+  enum {
+    offset = HE_STATUS1
+  };
+  uint64_t value;
+  struct {
+    uint32_t numPendReads : 32;
+    uint32_t numPendWrites : 32;
+  };
+};
+
+
+// ERROR
+union he_error {
+  enum {
+    offset = HE_ERROR
+  };
+  uint64_t value;
+  struct {
+    uint32_t error : 32;
+    uint32_t Rsvd : 32;
+  };
+};
+
+// STRIDE
+union he_stride {
+  enum {
+    offset = HE_STRIDE
+  };
+  uint32_t value;
+  struct {
+    uint32_t Stride : 32;
+  };
+};
+
+const std::map<std::string, uint32_t> he_modes = {
+  { "lpbk", HOST_EXEMODE_LPBK1},
+  { "read", HOST_EXEMODE_READ},
+  { "write", HOST_EXEMODE_WRITE},
+  { "trput", HOST_EXEMODE_TRUPT},
+};
+
+const std::map<std::string, uint32_t> he_req_cl_len= {
+  { "cl_1", HOSTEXE_CL_1},
+  { "cl_2", HOSTEXE_CL_2},
+  { "cl_3", HOSTEXE_CL_3},
+  { "cl_4", HOSTEXE_CL_4},
+};
+
+const std::map<std::string, uint32_t> he_test_mode = {
+  { "test_rollover", HOSTEXE_TEST_ROLLOVER},
+  { "test_termination", HOSTEXE_TEST_TERMINATION}
+};
+
+using test_afu = opae::afu_test::afu;
+using test_command = opae::afu_test::command;
+
+class host_exerciser : public test_afu {
+public:
+    host_exerciser()
+  : test_afu("host_exerciser")
+  , count_(1)
+  {
+    // Mode
+    app_.add_option("-m,--mode", he_modes_, "host exerciser mode {lpbk,read, write, trput}")
+      ->transform(CLI::CheckedTransformer(he_modes))->default_val("lpbk");
+
+    // Cache line
+    app_.add_option("--cl", he_req_cl_len_, "number of CLs per request{cl_1, cl_2, cl_3, cl_4}")
+       ->transform(CLI::CheckedTransformer(he_req_cl_len))->default_val("cl_1");
+
+    // Configures test rollover or test termination
+    app_.add_option("--ccont", he_ccont_, "Configures test rollover or test termination")->default_val(false);
```

Comment:
Oh, it is --ccont.  Just checking, is that what you intended?

Reply:
fixed